app/services/storage_migrator.rb

```diff
@@ -0,0 +1,49 @@
+# typed: false
+# frozen_string_literal: true
+
+# Move files into druid-based storage location
+class StorageMigrator
+  def self.migrate(druid:, source: default_source, target: default_target)
+    new(druid: druid, source: source, target: target)
+  end
+
+  def self.default_source
+    ActiveStorage::Service.configure(:local, active_storage_configs)
+  end
+  private_class_method :default_source
+
+  def self.default_target
+    ActiveStorage::Service.configure(:druid, active_storage_configs)
+  end
+  private_class_method :default_target
+
+  def self.active_storage_configs
+    # Need to load ActiveStorage::Blob to force the service configurations to load
+    ActiveStorage::Blob.connection
+    Rails.configuration.active_storage.service_configurations
+  end
+  private_class_method :active_storage_configs
+
+  def initialize(druid:, source:, target:)
+    @druid = druid
+    @source = source
+    @target = target
+  end
+
+  def migrate
+    ActiveStorage::Blob.service = source
+
+    attached_files = Work.find_by(druid: druid).attached_files
+    Rails.logger.info "Migrating #{attached_files.count} blobs to druid disk location for #{druid}..."
+    attached_files.each do |af|
+      Rails.logger.info "\t#{af.filename}"
+      af.blob.open do |blob|
+        target.upload(af.blob.key, blob, checksum: af.blob.checksum)
```

Comment:
Note that a checksum check is baked into this file move operation. Yay for ActiveStorage.